src/clj_foundation/templates.clj

```diff
@@ -130,7 +130,7 @@
 
   If :partial-resolve true is specified in options, resolves as many
   variables as possible.  Unresolved variables are returned as ${var-name}."
-  [parameter-source :- (types String clojure.lang.ISeq)
+  [parameter-source :- (types String clojure.lang.ASeq)
```

Comment:
The (types... schema needs concrete types.  (instance? IFoo val) isn't sufficient.